platform/darwin/src/MGLOfflineStorage+MGLURLSchemeTemplateAdditions.mm

```diff
@@ -0,0 +1,33 @@
+#import "MGLOfflineStorage+MGLURLSchemeTemplateAdditions.h"
+#import "MGLOfflineStorage_Private.h"
+
+@implementation MGLOfflineStorage (MGLURLSchemeTemplateAdditions)
+
+/**
+ Sets a URL template for the given scheme.
+
+ Any URLs encountered in styles and subsequently loaded sources will be fed through
+ a URL replacement mechanism that e.g. transforms `mapbox://` URLs into regular
+ `https://` URLs. You can provide URL replacement patterns for your own scheme,
+ e.g. `example://`.
+
+ Any query string will be preserved from the original URL and merged with a
+ query string of the template URL, if any.
+
+ @param tpl A templated URL. it may contain these brace-enclosed replacement
+    tokens, e.g. for `"http://example.com/foo/bar/baz.png"`, valid tokens are:
+    * `{scheme}`: `"http"`
+    * `{domain}`: `"example.com"`
+    * `{path}`: `"foo/bar/baz.png"` (without leading `/`)
```

Comment:
Why is the path without leading `/`? That is inconsistent with the definitions of path elsewhere, e.g. [`window.Location`](https://developer.mozilla.org/en-US/docs/Web/API/Location), [node](https://nodejs.org/api/url.html#url_urlobject_pathname), [RFC 3986](https://tools.ietf.org/html/rfc3986#section-3.3).